plaso/parsers/google_logging.py

```diff
@@ -0,0 +1,262 @@
+          'month_number'),
+      'day': text_parser.PyparsingConstants.ONE_OR_TWO_DIGITS.setResultsName(
+          'day'),
+      'hour': text_parser.PyparsingConstants.TWO_DIGITS.setResultsName(
+          'hour'),
+      'minute': text_parser.PyparsingConstants.TWO_DIGITS.setResultsName(
+          'minute'),
+      'second': text_parser.PyparsingConstants.TWO_DIGITS.setResultsName(
+          'second'),
+      'microsecond': pyparsing.Word(pyparsing.nums, exact=6).setParseAction(
+          text_parser.PyParseIntCast).setResultsName('microsecond'),
+      'thread_id': pyparsing.Word(pyparsing.nums).setResultsName('thread_id'),
+      'file_name':
+          (pyparsing.Word(pyparsing.printables.replace(':', '')).setResultsName(
+              'file_name')),
+      'line_number':
+          (pyparsing.Word(pyparsing.nums).setResultsName('line_number')),
+      'message':
+          (pyparsing.Regex('.*?(?=($|\n[IWEF][0-9]{4}))', re.DOTALL).
+           setResultsName('message'))}
+
+  _PYPARSING_COMPONENTS['date'] = (
+      _PYPARSING_COMPONENTS['month_number'] +
+      _PYPARSING_COMPONENTS['day'] +
+      _PYPARSING_COMPONENTS['hour'] + pyparsing.Suppress(':') +
+      _PYPARSING_COMPONENTS['minute'] + pyparsing.Suppress(':') +
+      _PYPARSING_COMPONENTS['second'] + pyparsing.Optional(
+          pyparsing.Suppress('.') +
+          _PYPARSING_COMPONENTS['microsecond']))
+
+  # Grammar for individual log event lines.
+  _LOG_LINE = (
+      _PYPARSING_COMPONENTS['priority'] + _PYPARSING_COMPONENTS['date']
+      + _PYPARSING_COMPONENTS['thread_id'] +
+      _PYPARSING_COMPONENTS['file_name'] + pyparsing.Suppress(':') +
+      _PYPARSING_COMPONENTS['line_number'] + pyparsing.Suppress('] ') +
+      _PYPARSING_COMPONENTS['message'] + pyparsing.lineEnd())
+
+  # Grammar for the log file greeting.
+  _GREETING = (
+      _PYPARSING_COMPONENTS['year'] + pyparsing.Suppress('/') +
+      _PYPARSING_COMPONENTS['month_number'] + pyparsing.Suppress('/') +
+      _PYPARSING_COMPONENTS['day'] + _PYPARSING_COMPONENTS['hour'] +
+      pyparsing.Suppress(':') + _PYPARSING_COMPONENTS['minute'] +
+      pyparsing.Suppress(':') + _PYPARSING_COMPONENTS['second'] +
+      pyparsing.Regex('.*?(?=($|\n[IWEF][0-9]{4}))', re.DOTALL) +
+      pyparsing.lineEnd())
+
+  _GREETING_START = 'Log file created at: '
+
+  # Our initial buffer length is the length of the string we verify with.
+  _INITIAL_BUFFER_SIZE = len(_GREETING_START)
+
+  # Once we're sure we're reading a valid file, we'll use a larger read buffer.
+  _RUNNING_BUFFER_SIZE = 5120
+
+  # Order is important here, as the structures are checked against each line
+  # sequentially, so we put the most common first, and the most expensive
+  # last.
+  LINE_STRUCTURES = [
+      ('log_entry', _LOG_LINE),
+      ('greeting_start', pyparsing.Literal(_GREETING_START)),
+      ('greeting', _GREETING)]
+
+  _SUPPORTED_KEYS = frozenset([key for key, _ in LINE_STRUCTURES])
```

Comment:
list comprehension is likely not needed here.
=> `_SUPPORTED_KEYS = frozenset(LINE_STRUCTURES.keys())`

Reply:
ack, I see, treacherous `, _`

Reply:
ack